app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,433 @@
 
 import _ from "lodash";
 import * as React from "react";
-import { Modal, Button, Input, Icon } from "antd";
-import update from "immutability-helper";
+import { Modal, Button, Tooltip, Icon, Table, InputNumber, Checkbox } from "antd";
+import * as Utils from "libs/utils";
 import { updateUser } from "admin/admin_rest_api";
-import type { APIUserType } from "admin/api_flow_types";
+import { handleGenericError } from "libs/error_handling";
+import type { APIUserType, ExperienceDomainListType } from "admin/api_flow_types";
+import SelectExperienceDomain from "components/select_experience_domain";
+
+const { Column } = Table;
+
+// TODO => visual bug when editing a value of a domain the first time !!!
+
+type MultipleUserTableEntry = {
+  domain: string,
+  value: number,
+  lowestValue: number,
+  highestValue: number,
+  isShared: boolean,
+};
+
+type SingleUserTableEntry = {
+  domain: string,
+  value: number,
+};
 
 type Props = {
-  onChange: Function,
-  onCancel: Function,
+  onChange: (Array<APIUserType>) => void,
+  onCancel: () => void,
   visible: boolean,
-  selectedUserIds: Array<string>,
-  users: Array<APIUserType>,
+  selectedUsers: Array<APIUserType>,
 };
 
 type State = {
-  domain: ?string,
-  level: ?string,
+  multipleUsersEntries: Array<MultipleUserTableEntry>,
+  singleUsersEntries: Array<SingleUserTableEntry>,
+  showOnlySharedExperiences: boolean,
 };
 
 class ExperienceModalView extends React.PureComponent<Props, State> {
-  state = {
-    domain: null,
-    level: null,
-  };
-
-  increaseExperience = (): void => {
-    this.setExperience(null, true);
-  };
-
-  setExperience = (event: ?SyntheticInputEvent<>, shouldAddValue: boolean = false): void => {
-    const { domain, level } = this.state;
-    if (domain && level !== null) {
-      const newUserPromises = this.props.users.map(user => {
-        if (this.props.selectedUserIds.includes(user.id)) {
-          let newExperienceLevel = parseInt(level);
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      multipleUsersEntries: this.loadMultipleUserEntries(props.selectedUsers),
+      singleUsersEntries: this.loadSingleUserEntries(props.selectedUsers),
+      showOnlySharedExperiences: false,
+    };
+  }
 
-          if (shouldAddValue && user.experiences[domain]) {
-            newExperienceLevel = user.experiences[domain] + parseInt(level);
-          }
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.visible && !this.props.visible) {
+      this.setState({
+        multipleUsersEntries: this.loadMultipleUserEntries(nextProps.selectedUsers),
+        singleUsersEntries: this.loadSingleUserEntries(nextProps.selectedUsers),
+      });
+    }
+  }
 
-          const newUser = update(user, {
-            experiences: { [domain]: { $set: newExperienceLevel } },
+  loadMultipleUserEntries = (users: Array<APIUserType>): Array<MultipleUserTableEntry> => {
```

Comment:
I find this method a bit hard to understand. Can you try to make more use of lodash's helper functions? For example, if you want to find out which experience domains are shared you can write something similar to:

`_.intersection(...users.map(user => Object.keys(user.experiences)))`

When you want to find out the min/max value for a certain domain you can leverage `_.min`/`_.max` etc.